src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -174,6 +177,47 @@ public int ExternalAttributes
             }
         }
 
+        /// <summary>
+        /// Gets or sets the optional entry comment.
+        /// </summary>
+        [AllowNull]
+        public string Comment
+        {
+            get
+            {
+                if (_fileComment == null)
+                {
+                    _fileComment = _fileCommentBytes != null ? DecodeBytesToString(_fileCommentBytes) : string.Empty;
+                }
+                return _fileComment;
+            }
+
+            set
+            {
+                // Reset it. Will re-acquire value in getter.
+                _fileComment = null;
+
+                if (value != null)
+                {
+                    bool isUTF8;
+                    _fileCommentBytes = ZipHelper.EncodeStringToBytes(value, _archive?.EntryNameAndCommentEncoding, out isUTF8);
+
+                    if (_fileCommentBytes.Length > ushort.MaxValue)
+                    {
+                        _fileCommentBytes = _fileCommentBytes[0..ushort.MaxValue];
+                    }
+                    if (isUTF8)
+                        _generalPurposeBitFlag |= BitFlagValues.UnicodeFileName;
+                    else
+                        _generalPurposeBitFlag &= ~BitFlagValues.UnicodeFileName;
```

Comment:
Looks to me like FileName and Comment behave as if they own this bit.  What happens when one is UTF8 and the other isn't and vice versa?  Seems like you could get in a bad state here based on order things are set.

Reply:
According to the spec, they do share it:

```
    Bit 11: Language encoding flag (EFS).  If this bit is set,
                the filename and comment fields for this file
                MUST be encoded using UTF-8. (see APPENDIX D)
```

Appendix D:
```
D.2 If general purpose bit 11 is unset, the file name and comment SHOULD conform 
to the original ZIP character encoding.  If general purpose bit 11 is set, the 
filename and comment MUST support The Unicode Standard, Version 4.1.0 or 
greater using the character encoding form defined by the UTF-8 storage 
specification.  The Unicode Standard is published by the The Unicode
Consortium (www.unicode.org).  UTF-8 encoded data stored within ZIP files 
is expected to not include a byte order mark (BOM). 
```

Reply:
Right, but as implemented you're assuming the value is independent.  So suppose you first set a UTF8 `Comment`, then set a non-UTF8 `FullName`, the setter for `FullName` will clear the bit, even though `Comment` needs it to be set.